apiserver/database/__init__.py

```diff
@@ -55,6 +57,10 @@ def initialize(cls):
         if override_port:
             log.info(f"Using override mongodb port {override_port}")
 
+        override_connection_string = getenv(OVERRIDE_CONNECTION_STRING_ENV_KEY)
```

Comment:
Since this overrides both `override_hostname` and `override_port`, I suggest checking for it first and not printing out other messages that might be confusing, something like:
```python
override_connection_string = getenv(OVERRIDE_CONNECTION_STRING_ENV_KEY)
if override_connection_string:
    log.info(f"Using override mongodb connection string {override_connection_string}")
else:
    override_hostname = first(map(getenv, OVERRIDE_HOST_ENV_KEY), None)
    if override_hostname:
        log.info(f"Using override mongodb host {override_hostname}")
    override_port = first(map(getenv, OVERRIDE_PORT_ENV_KEY), None)
    if override_port:
        log.info(f"Using override mongodb port {override_port}")

for key, alias in get_items(Database).items():

    ...

    if override_connection_string:
        entry.host = override_connection_string
    else:
        if override_hostname:
            entry.host = furl(entry.host).set(host=override_hostname).url
        if override_port:
            entry.host = furl(entry.host).set(port=override_port).url

```